client/rest-high-level/build.gradle

```diff
@@ -133,6 +133,7 @@ testClusters.integTest {
   setting 'xpack.security.authc.api_key.enabled', 'true'
   // Truststore settings are not used since TLS is not enabled. Included for testing the get certificates API
   setting 'xpack.security.http.ssl.certificate_authorities', 'testnode.crt'
+  setting 'xpack.security.http.ssl.verification_mode', 'certificate'
   setting 'xpack.security.transport.ssl.truststore.path', 'testnode.jks'
```

Comment:
This is "interesting". Because `verification_mode` now defaults to `none` - instead of `full` - (because `client_authentication` also defaults to `none`), the certs are not loaded, so they are not exposed in the `client.security().getSslCertificates`. The test otherwise fails with:
```
java.lang.AssertionError: 
Expected: <9>
     but: was <8>
	at __randomizedtesting.SeedInfo.seed([F216B34A8B4F4F2C:C1CDF65E57389ED7]:0)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.junit.Assert.assertThat(Assert.java:923)
	at org.elasticsearch.client.documentation.SecurityDocumentationIT.testGetSslCertificates(SecurityDocumentationIT.java:998)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:938)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:974)
```

Reply:
Given that `verification_mode` is `none` when `client_authentication` is `none` (so as not to trigger deprecation warnings in the default case), and that this leads to not loading the certificates is a slight behavior change. But I think this is not important?